File: typescript/darts/darts.ts
export function score(x: number, y: number): number {
  const distanceToCenter: number = Math.sqrt(Math.abs(x)**2 + Math.abs(y)**2);
  let score = 0;
  if (distanceToCenter <= 1) {
    score += 10;
  } else if (distanceToCenter <= 5) {
    score += 5;
  } else if (distanceToCenter <= 10) {
    score += 1;
  }
  return score;
}

File: ruby/darts/darts.rb
class Darts

  private
  attr_writer :distance_to_center

  def initialize(x, y)
    self.distance_to_center = Math.sqrt(x.abs**2 + y.abs**2)
  end

  public
  attr_reader :distance_to_center
  
  def score
    case distance_to_center
    when (0..1)
      10
    when (1..5)
      5
    when (5..10)
      1
    else
      0
    end
  end
end